hspec-discover/test/Test/Hspec/Discover/SortSpec.hs

```diff
@@ -0,0 +1,57 @@
+module Test.Hspec.Discover.SortSpec (main, spec) where
+
+import           Helper
+
+import           Data.List (sortBy)
+import           Data.Ord (comparing)
+
+import           Test.Hspec.Discover.Sort
+import           Test.QuickCheck
+
+main :: IO ()
+main = hspec spec
+
+
+spec :: Spec
+spec = do
+  describe "naturalSortKey" $ do
+    context "compares" $ do
+      it "reflexively" $ property $ \a -> do
+        a <=> a `shouldBe` EQ
+      it "symmetrically" $ property $ \a b -> do
+        a <=> b === compare EQ (b <=> a)
+      it "transitively" $ property $ \a b c -> do
+        a <=> b == b <=> c ==> a <=> b === a <=> c
```

Comment:
I think eflexivity, symmetry and transitivity is always given for derived instances, so there is no need to test for this.

We might want to test for injectivity though, see https://github.com/hspec/hspec/commit/62e0c4437c7fb88179491941081257b9cdbb8096.

Reply:
Good point. Thanks for merging and cleaning up 😃